src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -142,5 +190,11 @@ public int GetHashCode(Subscriber obj)
                 return (obj.Endpoint ?? string.Empty).Length;
             }
         }
+
+        internal class CacheItem
+        {
+            public DateTime Stored;
+            public Task<IEnumerable<Subscriber>> Subscribers;
```

Comment:
A question. Would storing awaited result be cheaper? In other words: the task that is returned from `GetSubscriptions` could have something captured in lambda/state machine. I'd assume that if we stored `Task.FromResult (await GetSubscriptions)` this could introduce less overhead. Am missing something?